newspaper/article.py

```diff
@@ -272,9 +271,12 @@ def fetch_images(self):
         if self.clean_top_node is not None and not self.has_top_image():
             first_img = self.extractor.get_first_img_url(
                 self.url, self.clean_top_node)
-            self.set_top_img(first_img)
+            if self.config.fetch_images:
+                self.set_top_img(first_img)
+            else:
+                self.set_top_img_no_check(first_img)
 
-        if not self.has_top_image():
+        if not self.has_top_image() and self.config.fetch_images:
```

Comment:
the meat of the fix, where most time is spent on some sites.

Reply:
👍